packages/remix/src/ssr/types.ts

```diff
@@ -35,7 +35,9 @@ export type RootAuthLoaderCallbackReturn =
   | Promise<Response>
   | Response
   | Promise<Record<string, unknown>>
-  | Record<string, unknown>;
+  | Record<string, unknown>
+  | Promise<null>
+  | null;
```

Comment:
👋🏻 
Going through the Remix rootAuthLoader code, I see that we currently do not allow `null` to be returned from the rootAuthLoader callback. Take a look at [this assertion](https://github.com/clerkinc/javascript/blob/6c095fba9d37b1e657f5a87f3c1c2e56e0ac45cd/packages/remix/src/ssr/rootAuthLoader.ts#L54) and [this error](https://github.com/clerkinc/javascript/blob/42e05494cb692c2056be020d720dfca66679ecea/packages/remix/src/errors.ts#L51)

As I recall, the reason for this is that we rely on receiving an object-like response in which we can inject the auth state of the app in order to render the interstitial page, if needed, from the client-side components.

Do we have any specific use-cases where return `null` would be required?

Reply:
Yep, I noticed the runtime checks! See the associated issue: #1615, though I'm not sure it makes sense for us to support it given that it's not just a type issue and we rely on an object being returned.

Alternatively, we could handle this case internally and make sure we return an object with the injected auth state even if nothing is returned.

Reply:
@nikosdouvlis To improve DX, I've opted to allow returning `null` from the loader callback and have updated the logic to handle creating an empty response accordingly. Aligning with what Remix allows seems like a good choice here. Remix's error:

> Error: You defined a loader for route "root" but didn't return anything from your `loader` function. Please return a value or `null`.

Reply:
Thank you Bryce, this makes sense.

This sounds like an edge scenario but I'd still like to make sure we are all aligned. In the following example, what would be the return value for the `[useLoaderData](https://remix.run/docs/en/main/hooks/use-loader-data)` hook? 

```
export const loader: LoaderFunction = args => {
  return rootAuthLoader(args, ({ request }) => {
    const { userId } = request.auth;
    return userId ? moviesRepo.getMoviesFor(userId) : null;
  });
};

function App() {
  const movies = useLoaderData();
  return (
    <html lang='en'>
     ...
}
```

Of course, the ideal way to right this would be to either use `[]` as a fallback value or wrap the return in a `{ movies: ... }` object.

Reply:
Bryce and I discussed about this offline, we're going to proceed with the changes shown in the PR :)